diesel/src/expression_methods/global_expression_methods.rs

```diff
@@ -434,6 +434,80 @@ pub trait ExpressionMethods: Expression + Sized {
     fn asc(self) -> dsl::Asc<Self> {
         Asc::new(self)
     }
+
+    /// Converts this potentially nullable expression into one which will be **assumed**
+    /// to be not-null. This method has no impact on the generated SQL, however it will
+    /// enable you to attempt deserialization of the returned value in a non-`Option`.
+    ///
+    /// This is meant to cover for cases where you know that given the `WHERE` clause
+    /// the field returned by the database will never be `NULL`.
+    ///
+    /// This **will cause runtime errors** on `load()` if the "assume" turns out to be incorrect.  
+    /// If incorrect in `Nullable` tuples, this **will cause silent fails** (see second
+    /// example).
+    ///
+    /// # Examples
+    /// ## Normal usage
+    /// ```rust
+    /// # #![allow(dead_code)]
+    /// # include!("../doctest_setup.rs");
+    /// # use diesel::sql_types::*;
+    /// #
+    /// table! {
+    ///     posts {
+    ///         id -> Integer,
+    ///         user_id -> Integer,
+    ///         author_name -> Nullable<VarChar>,
+    ///     }
+    /// }
+    ///
+    /// fn main() {
+    ///     use self::posts::dsl::*;
+    ///     let connection = &mut establish_connection();
+    ///
+    ///     let data = posts
+    ///         .filter(author_name.is_not_null())
+    ///         .select(author_name.assume_not_null())
+    ///         .load::<String>(connection);
+    ///     println!("{:?}", data);
+    /// }
+    /// ```
+    ///
+    /// ## Dangerous usage - use only if you're sure you know what you're doing!
+    /// ```rust
+    /// # #![allow(dead_code)]
+    /// # include!("../doctest_setup.rs");
+    /// # use diesel::sql_types::*;
+    /// # use schema::users;
+    /// #
+    /// table! {
+    ///     posts {
+    ///         id -> Integer,
+    ///         user_id -> Integer,
+    ///         author_name -> Nullable<VarChar>,
+    ///     }
+    /// }
+    /// #
+    /// # joinable!(posts -> users (user_id));
+    /// # allow_tables_to_appear_in_same_query!(posts, users);
+    ///
+    /// fn main() {
+    ///     use self::posts;
+    ///     use self::users;
+    ///     let connection = &mut establish_connection();
+    ///
+    ///     let data = posts::table.left_join(users::table)
+    ///         .select((users::id, posts::author_name.assume_not_null()).nullable())
+    ///         .load::<Option<(i32, String)>>(connection);
+    ///     // This will cause the `Option` to be `None` where the `left_join` succeeded but the
+    ///     // `author_name` turned out to be `NULL`, because `Option` deserialization relies
+    ///     // on encountering an unexpected `NULL` while deserializing the inner fields.
```

Comment:
I believe this documentation belongs to the `FromSqlRow` impl of `Option<T>` instead, as this can also happen in other constructed cases. I would rather include an example that really returns an error from `load()`

Reply:
I'm not sure where we could put that comment about `FromSqlRow`.
Also it seems important to me to underline this behavior implied by the usage of `assume_not_null()`, as it's the one that feels the most of a trap to me: if you were wrong about "assuming" that `author_name` (unrelated to the left join) was not null, then instead of having the option set iff the left join succeeded, it will be `None` when it was "assume"d wrong.

Reply:
> I'm not sure where we could put that comment about FromSqlRow... ?
Also it seems important to me to underline this behavior implied by the usage of assume_not_null(), as it's the one that feels the most of a trap to me: if you were wrong about "assuming" that author_name (unrelated to the left join) was not null, then instead of having the option set iff the left join succeeded, it will be None when it was "assume"d wrong.

We could put that directly on the implementation of `FromSqlRow for Option<T>` and link that from here. Would that be an option?

Reply:
So if I understand correctly that would actually be [here](https://docs.diesel.rs/master/diesel/deserialize/trait.FromStaticSqlRow.html#impl-FromStaticSqlRow%3CNullable%3C(SA%2C)%3E%2C%20__DB%3E-for-Option%3C__T%3E) with the documentation repeated for every tuple size, as that is actually defined here:
https://github.com/diesel-rs/diesel/blob/cfb95067153ba55dff556c0b580cbc0e47f3fd28/diesel/src/type_impls/tuples.rs#L259-L287

Or maybe we want to add it directly on the `FromStaticSqlRow` or `Queryable` trait?

Reply:
That's a good point. Adding it to the general documentation of `Queryable` seems a good solution. (I would add it there, as that's the trait most users interact with.)

Reply:
a7fae660e9